kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -133,19 +134,23 @@ private static Schema inferSchemaInternal(String name, InputStream incoming,
     boolean[] nullable = new boolean[header.length];
     boolean[] empty = new boolean[header.length];
 
-    for (int processed = 0; processed < DEFAULT_INFER_LINES; processed += 1) {
+    for (int processed = 0; processed < props.linesToSample; processed += 1) {
       if (line == null) {
         break;
       }
 
       for (int i = 0; i < header.length; i += 1) {
         if (i < line.length) {
-          if (types[i] == null) {
-            types[i] = inferFieldType(line[i]);
+          Schema.Type prevType = types[i];
+          Schema.Type currType = inferFieldType(line[i]);
+          if (prevType == null) {
+            types[i] = currType;
             if (types[i] != null) {
-              // keep track of the value used
               values[i] = line[i];
             }
+          } else if (promoteType(prevType, currType)) {
```

Comment:
I've changed the name as you suggested. The code I've used in the CSVUtil mergeTypes method is similar to the same method in the SchemaUtil file, but I've kept the one I've written because it handles merging of previous numeric types to strings when necessary. I do so because CSV files can be notorious for having odd string flags in otherwise numeric columns, so I try to allow for strings as a 'most generic' type when necessary. Let me know if you disagree with this reasoning, I could rearrange it to make use of the other mergeTypes method if its is preferred.